Redefined specialization in type_traits.h using MSVC (Protobuf 3.1.0)

This commit added [explicit specializations](https://github.com/google/protobuf/blame/b9bad65bf32cebc6210b4da9251919503d479dbd/src/google/protobuf/stubs/type_traits.h#L146) for `is_integral<__int64>` and `is_integral<unsigned __int64` when compiling with Visual C++: https://github.com/google/protobuf/commit/3c88e1b53de832547e04f2e1c075bb1aae0ca0b4

It seems that by default `HAVE_LONG_LONG` is not defined on this platform, so these specializations do not collide with the specializations for `is_integral<long long>` and `is_integral<unsigned long long>` [immediately above](https://github.com/google/protobuf/blame/b9bad65bf32cebc6210b4da9251919503d479dbd/src/google/protobuf/stubs/type_traits.h#L142) them.

However, other projects use `HAVE_LONG_LONG` on Windows, which leads to an error about the specialization already being defined. In particular, when I include `<Python.h>` from Anaconda for Windows before a header that includes some protobuf headers, I get the following compilation error:

```
[...]google/protobuf/stubs/type_traits.h(151): error C2766: explicit specialization; 'google::protobuf::internal::is_integral<__int64>' has already been defined
[...]google/protobuf/stubs/type_traits.h(143): note: see previous definition of 'is_integral<__int64>'
[...]google/protobuf/stubs/type_traits.h(152): error C2766: explicit specialization; 'google::protobuf::internal::is_integral<unsigned __int64>' has already been defined
[...]google/protobuf/stubs/type_traits.h(144): note: see previous definition of 'is_integral<unsigned __int64>'
```

Here's a simple repro, `test.cc`:

``` c++
//#include <Python.h>  // The following #define has the same effect.
#define HAVE_LONG_LONG
#include "google/protobuf/stubs/type_traits.h"

int main(int argc, char** argv) {
  return 0;
}
```

Reproduced using the following command line with Visual Studio 2015 Update 1:

```
C:\temp> cl.exe test.cc /I<path\to\protobuf\src\>
```

Could the conditional compilation be refactored to avoid this problem? @xfxyjwf, since you made the commit, any ideas?

--- a/src/google/protobuf/stubs/type_traits.h
+++ b/src/google/protobuf/stubs/type_traits.h
@@ -139,17 +139,9 @@ template<> struct is_integral<int> : true_type { };
 template<> struct is_integral<unsigned int> : true_type { };
 template<> struct is_integral<long> : true_type { };
 template<> struct is_integral<unsigned long> : true_type { };
-#ifdef HAVE_LONG_LONG
+#if defined(HAVE_LONG_LONG) || defined(_MSC_VER)
 template<> struct is_integral<long long> : true_type { };
 template<> struct is_integral<unsigned long long> : true_type { };
-#endif
-#if defined(_MSC_VER)
-// With VC, __int8, __int16, and __int32 are synonymous with standard types
-// with the same size, but __int64 has not equivalent (i.e., it's neither
-// long, nor long long and should be treated differnetly).
-// https://msdn.microsoft.com/en-us/library/29dh1w7z.aspx
-template<> struct is_integral<__int64> : true_type { };
-template<> struct is_integral<unsigned __int64> : true_type {};
 #endif
 template <class T> struct is_integral<const T> : is_integral<T> { };
 template <class T> struct is_integral<volatile T> : is_integral<T> { };

